frontends/ast/genrtlil.cc

```diff
@@ -1511,14 +1573,38 @@ RTLIL::SigSpec AstNode::genRTLIL(int width_hint, bool sign_hint)
 				}
 				if (child->type == AST_ARGUMENT) {
 					RTLIL::SigSpec sig;
-					if (child->children.size() > 0)
+					RTLIL::SigSpec new_sig;
+					if (child->children.size() > 0) {
 						sig = child->children[0]->genRTLIL();
+						for (int i = 0; i < GetSize(sig); i++) {
+							if (sig[i].wire) {
```

Comment:
This assumes that all module ports are outputs (drivers), which of course isn't true. This raises the question: Maybe it would be better to simply tag all wor/wand wires with a `wor`/`wand` attribute, and insert the OR/AND cells in `hierarchy`, where we have the information which cell port is an input or output.

Reply:
Correct. Am I correct to assume that at this step (AST->RTLIL) there's no way to get the direction information? It seemed like that to me and that's why I came up with this simplifying assumption in the first place.

I'd be happy to move it into `hierarchy`, in case it would be an acceptable departure from the separation of 'verilog implementation details' from the RTLIL design.

Reply:
Yes, it's impossible to make the input/output information available at this step because it's possible that the verilog code for the module instantiated is being read later.

It's acceptable to have stuff like that in `hierarchy` when it's necessary to put it there. We already have similar things in `hierarchy` to deal with positional arguments in module instantiations (the cell ports are initially named $1, $2, $3, ... and so forth and then renamed in `hierarchy`) and array instantiations (for which we need to know the width of cell ports, which also isn't available in AST world).